server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/RestoreBuiltInAction.java

```diff
@@ -55,9 +59,15 @@ public void define(WebService.NewController controller) {
 
   @Override
   public void handle(Request request, Response response) {
+    verifyAdminPermission();
+
     String language = request.mandatoryParam("language");
-    service.restoreBuiltInProfilesForLanguage(language);
+    reset.resetLanguage(language);
     response.noContent();
   }
 
+  private void verifyAdminPermission() {
+    userSession.checkLoggedIn();
+    userSession.checkPermission(GlobalPermissions.QUALITY_PROFILE_ADMIN);
```

Comment:
Nice to have : it could have been good to add unit test on permission check (fail when not logged, fail when missing correct permission)